sqsworkers/crew.py

```diff
@@ -157,25 +158,41 @@ def poll_queue(self):
                 MessageAttributeNames=['All'],
                 MaxNumberOfMessages=self.max_number_of_messages,
                 WaitTimeSeconds=self.wait_time)
-
             if len(messages) > 0:
-                for message in messages:
-                    self.logger.info('processing %s messages %s' % (len(messages), messages))
-                    processor = self.crew.MessageProcessor(message)
-                    self.crew.statsd.increment('process.record.start', 1, tags=[])
-                    processed = processor.start()
-                    if processed:
-                        deleted = self.queue.delete_messages(
-                            Entries=[{
-                                'Id': message.message_id,
-                                'ReceiptHandle': message.receipt_handle
-                            }]
-                        )
-                        self.crew.statsd.increment('process.record.success', 1, tags=[])
-                        self.logger.info('%s messages processed successfully and deleted %s' % (len(messages), deleted))
-                    else:
-                        self.crew.statsd.increment('process.record.failure', 1, tags=[])
-
+                try:
+                    for message in messages:
+                        self.logger.info('processing %s messages %s' % (len(messages), messages))
+                        processor = self.crew.MessageProcessor(message)
+                        self.crew.statsd.increment('process.record.start', 1, tags=[])
+                        processed = processor.start()
+                        if processed:
+                            deleted = self.queue.delete_messages(
+                                Entries=[{
+                                    'Id': message.message_id,
+                                    'ReceiptHandle': message.receipt_handle
+                                }]
+                            )
+                            self.crew.statsd.increment('process.record.success', 1, tags=[])
+                            self.logger.info('%s messages processed successfully and deleted %s' % (len(messages), deleted))
+                        else:
+                            self.crew.statsd.increment('process.record.failure', 1, tags=[])
+                except Exception as e:
+                    self.excepton_handler(e, message, {'crew.name': self.name})
+                    # continue with the next message and do not delete
+                    pass
+
+    def exception_handler(self, e, message, extra_info):
```

Comment:
Can we have a basic exception handler and maybe have this a sample function which then can be imported and used by our projects which we know have all of these entities?